application/src/main/java/org/togetherjava/tjbot/commands/help/HelpSystemHelper.java

```diff
@@ -363,23 +363,25 @@ private void executeNoActivityAdviceCheck(long threadChannelId, long authorId) {
                     threadChannelId, authorId);
             return;
         }
-        if (threadChannel.getHistory()
-            .getRetrievedHistory()
-            .stream()
-            .map(Message::getAuthor)
-            .map(User::getIdLong)
-            .anyMatch(id -> id == authorId)) {
+        if (noAuthorActivity(authorId, threadChannel)) {
             MessageEmbed embed = HelpSystemHelper.embedWith(
                     """
-                            Hey there {}👋 It has been five minutes and you have not shown any activity in this thread.
-                            Please elaborate on your question, or close the thread if it was created accidentally, thanks 🙂
-                            """,
-                    User.fromId(authorId).getAsMention());
-            MessageCreateData message = new MessageCreateBuilder().setEmbeds(embed).build();
-            threadChannel.sendMessage(message).queue();
-
+                            Hey there %s👋  It has been a bit after you created this thread and you still did not share any details of your question.
+                            Helpers have seen your question already and are just waiting for you to elaborate on your problem and provide detailed information on it 👌
+                            If this thread was created accidentally, you can close it.
+                            """
+                        .formatted(User.fromId(authorId).getAsMention()));
+            threadChannel.sendMessageEmbeds(embed).queue();
         }
+    }
 
+    private static boolean noAuthorActivity(long authorId, ThreadChannel threadChannel) {
+        return threadChannel.getIterableHistory()
+            .stream()
+            .limit(10)
+            .map(Message::getAuthor)
+            .map(User::getIdLong)
+            .noneMatch(id -> id == authorId);
```

Comment:
if the limit is reached, it should assume the author send something. right now you assume there is no author activity, but that could be wrong (maybe the authors message is the 11th)

so if the limit is reached, fallback to assuming the author did send sth

Reply:
You're right, if the author had not posted anything the other helpers wouldbhave tagged them anyway.

But how would I implement that with a single stream chain?

Reply:
i doubt u can. gotta make it a tad more ugly 😛

Reply:
I have used `MessageHistory#size` I hope that does not result in an expensive network call, on first glance it did not seem like it did